GetStream/stream-chat-android
# [4305] Remove ThreeTenABP dependency

### 🎯 Goal

Closes https://github.com/GetStream/stream-chat-android/issues/4305

Remove ThreeTenABP dependency as it is barely used.

### 🧪 Testing

- Run the app and ensure that dates are formatted correctly. 
- Ensure that tests in `DefaultDateFormatterTest` make sense.

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
